From 3d02a1bb2d144ee5e671f725a82404111135588c Mon Sep 17 00:00:00 2001
From: "Teli,ShivaniX" <shivanix.teli@intel.com>
Date: Wed, 30 Nov 2022 12:31:48 +0530
Subject: [PATCH] Added BAD_VALUE mSensors.size() is zero

For celadon added BAD_VALUE mSensors.size() is zero
Fix for VtsHalSensorsV2_0TargetTest

Tracked-On: OAM-105026
Signed-off-by: Teli,ShivaniX <shivanix.teli@intel.com>

diff --git a/sensors/2.0/iiohal_mediation_v2.0/Sensors.h b/sensors/2.0/iiohal_mediation_v2.0/Sensors.h
index f33faee..c3b1edc 100644
--- a/sensors/2.0/iiohal_mediation_v2.0/Sensors.h
+++ b/sensors/2.0/iiohal_mediation_v2.0/Sensors.h
@@ -98,6 +98,9 @@ struct Sensors : public ISensorsInterface, public ISensorsEventCallback {
     }
 
     Return<Result> setOperationMode(OperationMode mode) override {
+	if(mSensors.size() <= 0) {
+            return Result::BAD_VALUE;
+	}
         for (auto sensor : mSensors) {
             sensor.second->setOperationMode(mode);
         }
-- 
2.38.1

